internal/services/storage/storage_share_resource.go

```diff
@@ -143,239 +150,424 @@ func resourceStorageShare() *pluginsdk.Resource {
-		log.Printf("[WARN] Unable to determine Account %q for Storage Share %q - assuming removed & removing from state", id.AccountId.AccountName, id.ShareName)
-		d.SetId("")
 		return nil
 	}
 
-	// The files API does not support bearer tokens (@manicminer, 2024-02-15)
-	client, err := storageClient.FileSharesDataPlaneClient(ctx, *account, storageClient.DataPlaneOperationSupportingOnlySharedKeyAuth())
+	id, err := fileshares.ParseShareID(d.Id())
 	if err != nil {
-		return fmt.Errorf("building File Share Client for %s: %+v", account.StorageAccountId, err)
+		return err
 	}
 
-	props, err := client.Get(ctx, id.ShareName)
+	existing, err := sharesClient.Get(ctx, *id, fileshares.DefaultGetOperationOptions())
 	if err != nil {
-		return err
+		if response.WasNotFound(existing.HttpResponse) {
+			log.Printf("[DEBUG] %q was not found, removing from state", *id)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving %q: %v", *id, err)
 	}
-	if props == nil {
-		log.Printf("[DEBUG] File Share %q was not found in %s - assuming removed & removing from state", id.ShareName, account.StorageAccountId)
-		d.SetId("")
-		return nil
+
+	d.Set("storage_account_id", commonids.NewStorageAccountID(id.SubscriptionId, id.ResourceGroupName, id.StorageAccountName).ID())
+
+	if model := existing.Model; model != nil {
+		d.Set("name", id.ShareName)
+		if props := model.Properties; props != nil {
+			d.Set("quota", props.ShareQuota)
+			// Resource Manager treats nil and "SMB" as the same and we may not get a full response here
+			enabledProtocols := fileshares.EnabledProtocolsSMB
+			if props.EnabledProtocols != nil {
+				enabledProtocols = *props.EnabledProtocols
+			}
+			d.Set("enabled_protocol", string(enabledProtocols))
+			d.Set("access_tier", string(pointer.From(props.AccessTier)))
+			d.Set("acl", flattenStorageShareACLs(pointer.From(props.SignedIdentifiers)))
+			d.Set("metadata", FlattenMetaData(pointer.From(props.Metadata)))
+		}
 	}
 
-	d.Set("name", id.ShareName)
-	d.Set("storage_account_name", id.AccountId.AccountName)
-	d.Set("quota", props.QuotaGB)
-	d.Set("url", id.ID())
-	d.Set("enabled_protocol", string(props.EnabledProtocol))
+	d.Set("resource_manager_id", id.ID())
 
-	accessTier := ""
-	if props.AccessTier != nil {
-		accessTier = string(*props.AccessTier)
+	account, err := storageClient.FindAccount(ctx, subscriptionId, id.StorageAccountName)
+	if err != nil {
+		return fmt.Errorf("retrieving Account %q for Share %q: %v", id.StorageAccountName, id.ShareName, err)
+	}
+	if account == nil {
+		return fmt.Errorf("locating Storage Account %q", id.StorageAccountName)
 	}
-	d.Set("access_tier", accessTier)
 
-	if err := d.Set("acl", flattenStorageShareACLs(props.ACLs)); err != nil {
-		return fmt.Errorf("flattening `acl`: %+v", err)
+	// Determine the file endpoint, so we can build a data plane ID
+	endpoint, err := account.DataPlaneEndpoint(client.EndpointTypeFile)
+	if err != nil {
+		return fmt.Errorf("determining File endpoint: %v", err)
```

Comment:
Shouldn't we **not** be using any Data Plane functionality here?

Reply:
This is unfortunately still necessary due to a dependency on `giovanni` until that functionality is ported to `go-azure-sdk`. This doesn't actually make any external calls to the data plane, should be fine to be used here until for now.